rewrite-java/src/main/java/org/openrewrite/java/UseStaticImport.java

```diff
@@ -128,35 +114,31 @@ public Javadoc visitReference(Javadoc.Reference reference, ExecutionContext ctx)
         }
     }
 
-    private static boolean methodNameConflicts(String methodName, Cursor cursor) {
+    private static boolean hasConflictingImport(String typeName, String methodName, Cursor cursor) {
         J.CompilationUnit cu = cursor.firstEnclosing(J.CompilationUnit.class);
         if (cu != null) {
             for (J.Import imp : cu.getImports()) {
-                if (imp.isStatic() && methodName.equals(imp.getQualid().getSimpleName())) {
+                if (imp.isStatic() &&
+                        methodName.equals(imp.getQualid().getSimpleName()) &&
+                        !typeName.equals(imp.getTypeName())) {
                     return true;
                 }
             }
         }
+        return false;
+    }
 
+    private static boolean hasConflictingMethod(String methodName, Cursor cursor) {
         Cursor cdCursor = cursor.dropParentUntil(it -> it instanceof J.ClassDeclaration || it == Cursor.ROOT_VALUE);
         Object maybeCd = cdCursor.getValue();
         if (!(maybeCd instanceof J.ClassDeclaration)) {
             return false;
         }
-        J.ClassDeclaration cd = (J.ClassDeclaration) maybeCd;
-        JavaType.FullyQualified ct = cd.getType();
-        if (ct == null) {
-            return false;
-        }
-
-        if (methodNameConflicts(methodName, ct)) {
-            return true;
-        }
-
-        return methodNameConflicts(methodName, cdCursor);
+        return hasConflictingMethod(methodName, ((J.ClassDeclaration) maybeCd).getType()) ||
+                hasConflictingMethod(methodName, cdCursor);
```

Comment:
The above could be flattened to just this; renaming the method makes intention clearer: we're looking for methods from super/interface/declared.